### PR TITLE
Category manager - banner item count

### PR DESCRIPTION
**Pull request by pe7er**
This patch creates in "**Category Manager: Banners**" 4 new columns with the number of banners per category. It contains a helper file for com_banners that is used in com_categories to calculate the number of banner items (published, unpublished, archived, and trashed) in each category.
## Test instructions:

The **Category Manager: Banners** (Components > Banners > Categories) does not display the number of banners in each category.

screen shot 2015-05-13 at 14 24 31

Create some new banners that are unpublished, trashed & archived (tip: name the unpublished banner "Unpublished Banner" to test if the calculations are correct).

screen shot 2015-05-13 at 14 24 31

**Note** This patch works only **in combination with** #6916. Install both #6916 and this patch (the order of installation does not matter, you just need both).
After installing the patch, The **Category Manager: Banners** (Components > Banners > Categories) should displays 4 new columns.

screen shot 2015-05-13 at 14 24 31
### Please also check the other administrator template **Hathor**

In Extensions > Template Manager > Select Hathor
Before the patch:

screen shot 2015-05-13 at 14 24 31
After the patch:
screen shot 2015-05-13 at 14 24 31

### CODE REVIEW

File: administrator/components/com_banners/helpers/countitems.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
+<?php
+/**
+ * @package     Joomla.Administrator
+ * @subpackage  com_content
```
Comment (losedk) [inline administrator/components/com_banners/helpers/countitems.php]: Should be com_banners

File: administrator/components/com_banners/helpers/countitems.php [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+defined('_JEXEC') or die;
+
+/**
+ * Contentitem component helper.
+ *
+ * @since  1.6
+ */
+class ContentitemsHelper extends JHelperContent
+{
+	public static $extension = 'com_content';
```
Comment (losedk) [inline administrator/components/com_banners/helpers/countitems.php]: Should this be com_banners?

**Comment by pe7er** [comment]
Thanks @losedk !
I've corrected it and uploaded a new patch.

**Comment by hans2103** [comment]
@test succes

Tested both desktop and mobile.

**Comment by woluweb** [comment]
on issue #6934 (count for BANNERS), I noticed that clicking on "zeros" would still show all banners of the category, no matter their "status".
But apparently, this little bug only occurs for com_banners (works fine for the current com_content, for com_contact and com_newsfeed)
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6934">issues.joomla.org/joomla-cms/6934</a>.</sub>

**Comment by pe7er** [comment]
Thank you both for testing. I've analysed & reproduced the problem: The status filter of the com_banners component works differently from other components.

The category manager com_categories creates number counters with hyperlinks with status filter **&filter[published]=**. All components listen for the "Select Status" filter to **&filter[published]=** (and expect a statuscode of 0,1,2 and -2). 

However the banner component listens to **&filter[state]=**.
What's the best way to "catch"  the variable **filter[published]** and set the **filter[state]**?

**Comment by woluweb** [comment]
Hi Peter,
Your question is a bit too technical for me.
Good sense would say : if the Banner Component is the only one to use [state] instead of [published], maybe we should fix the Banner Component instead of changing "state" by "published" on the fly ?
But maybe this is a stupid suggestion, I am not a developer :-)

**Comment by pe7er** [comment]
I've changed the 'state' field to 'published' in the filter administrator/components/com_banners/models/forms/filter_banners.xml and in the banners model. 

Now this PR (in combination with PR #6916) displays a working "banner item counts" in the **Category Manager: Banners** that are hyperlinked correctly to ** Banner Manager: Banners** with filtered items on **published** & **category** filter.

**Comment by woluweb** [comment]
Hi Peter,
I have just tested.
I confirm : the filters do now work perfectly also for Banner Component
Txs<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6934">issues.joomla.org/joomla-cms/6934</a>.</sub>

**Comment by Bakual** [comment]
Merged with https://github.com/joomla/joomla-cms/commit/28574c58cceccf59361fc12ad0fc1af2353831d8 into `3.5-dev`.

Thanks!

**Pull request closed without merging**